geode-core/src/main/java/org/apache/geode/cache/query/internal/index/AbstractIndex.java

```diff
@@ -2013,7 +2014,7 @@ synchronized void setPdxStringFlag(Object key) {
       return;
     }
     if (!this.isIndexedPdxKeys) {
-      if (key instanceof PdxString) {
+      if (key instanceof PdxString && ((InternalRegion) this.region).getCompressor() == null) {
```

Comment:
To avoid the cast you can change it to:
  region.getAttributes().getCompressor()

Reply:
Excellent. Thanks. I'll fix that.